providers/implementations/keymgmt/ml_kem_kmgmt.c.in

```diff
@@ -392,39 +398,48 @@ static int ml_kem_key_fromdata(ML_KEM_KEY *key,
      * When a private key is provided, without a seed, any public key also
      * provided will be ignored (apart from length), just as with the seed.
      */
-    if (include_private) {
-        /*
-         * When a seed is provided, the private and public keys may be ignored,
-         * after validating just their lengths.  Comparing encodings or hashes
-         * when applicable is possible, but not currently implemented.
-         */
-        p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_ML_KEM_SEED);
-        if (p != NULL
-            && OSSL_PARAM_get_octet_string_ptr(p, &seedenc, &seedlen) != 1)
-            return 0;
-        if (seedlen != 0 && seedlen != ML_KEM_SEED_BYTES) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_SEED_LENGTH);
-            return 0;
-        }
-        p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_PRIV_KEY);
-        if (p != NULL
-            && OSSL_PARAM_get_octet_string_ptr(p, &prvenc, &prvlen) != 1)
-            return 0;
-        if (prvlen != 0 && prvlen != v->prvkey_bytes) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_KEY_LENGTH);
-            return 0;
-        }
-    }
+    for (p = params; p->key != NULL; p++)
+        switch (ml_kem_key_types_find_pidx(p->key)) {
+        default:
+            break;
+
+        case PIDX_PKEY_PARAM_ML_KEM_SEED:
+            if (include_private) {
+                /*
+                 * When a seed is provided, the private and public keys may be ignored,
+                 * after validating just their lengths.  Comparing encodings or hashes
+                 * when applicable is possible, but not currently implemented.
+                 */
```

Comment:
This comment is somewhat outdated, the private key is checked against the seed when both are provided.
In the unexpected and unlikely case that a seed is provided along with only the public key no consistency check is performed (this does not currently arise when parsing PKCS#8 inputs), but perhaps a check should be made?

In any case the comment could be updated to match the code.